timestep.c

```diff
@@ -772,41 +751,106 @@ void rebuild_activelist(void)
  * function will drift to this moment, generate an output, and then
  * resume the drift.
  */
-inttime_t find_next_kick(inttime_t Ti_Current)
+inttime_t find_next_kick(inttime_t Ti_Current, int minTimeBin)
 {
-    /* Note that on startup, P[i].TimeBin == 0 for all particles,
-     * all bins except the zeroth are inactive and so we return 0 from this function.
-     * This ensures we run the force calculation for the first timestep.*/
-    /* find the smallest active bin*/
-    int n;
-    for(n = 0; n < TIMEBINS; n++)
-    {
-        if(TimeBinCount[n] > 0)
-            break;
-    }
-    MPI_Allreduce(MPI_IN_PLACE, &n, 1, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
     /* Current value plus the increment for the smallest active bin. */
-    return Ti_Current + dti_from_timebin(n);
+    return Ti_Current + dti_from_timebin(minTimeBin);
 }
 
 /* mark the bins that will be active before the next kick*/
 int update_active_timebins(inttime_t next_kick)
```

Comment:
This function perhaps shall take MinTimeBin sounded like anything below mintimbin will be activate.

Reply:
?

Reply:
Sorry I typed this in a rush. I mean the logic of update_active_timebins can be simplified if minTimeBin is passed in. The code apparently is just trying to rebuild that information from next_kick, which is inferred from minTimeBin.

Reply:
It can be simplified out of existence